source/common/access_log/access_log_impl.cc

```diff
@@ -255,6 +257,37 @@ Grpc::Status::GrpcStatus GrpcStatusFilter::protoToGrpcStatus(
   return static_cast<Grpc::Status::GrpcStatus>(status);
 }
 
+MetadataFilter::MetadataFilter(const envoy::config::accesslog::v3::MetadataFilter& filter_config)
+    : matcher_config_(filter_config.matcher()) {
+  default_res_ = true;
+  if (filter_config.has_no_key_default()) {
+    default_res_ = filter_config.no_key_default().value();
+  }
+}
+
+bool MetadataFilter::evaluate(const StreamInfo::StreamInfo& info, const Http::RequestHeaderMap&,
+                              const Http::ResponseHeaderMap&,
+                              const Http::ResponseTrailerMap&) const {
+
+  auto matcher = Matchers::MetadataMatcher(matcher_config_);
```

Comment:
Can we pre-construct this here and below to avoid doing this every evaluation?

Also, is there a way to modify the matcher to only match once and somehow return whether it's present or not? It seems not great that we have to match twice.

Reply:
Sure, that makes more sense.

No, the metadata matcher will return false if the key is not present for a regular value match. I could implement the matcher logic myself to avoid matching twice.

Reply:
Should I implement the matcher logic here, then? Or should I leave the two matchers?

Reply:
I don't have a strong opinion. Whatever you think is best.